### PR TITLE
Build: Add Vitest integration tests to sandboxes

### LINKED ISSUE #28707: Add Storybook Vitest plugin to Vite based (internal) sandboxes in the monorepo

### PR DESCRIPTION
**Pull request by yannbf**
Closes #28707

## What I did

This PR introduces `vitest` and the `@storybook/experimental-vitest-plugin` package to Vite based sandboxes.
It adds a new task called `vitest-integration` which you can run like:
```
yarn task --task vitest-integration --template react-vite/default-ts
```

Upon creating a sandbox, you can also run `yarn vitest` in the sandbox directory.

This task is run in CI for 10 sandboxes so far (will increase once we add Next.js with the Vite plugin).

Currently, the config.yml file has most of the tasks commented out so we can use `ci:daily` to test the sandboxes without spending too much money.

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

This PR introduces Vitest integration for Vite-based sandboxes, enhancing the testing capabilities of the Storybook project.

- Added `vitest-integration` task in `.circleci/config.yml` for CI testing.
- Updated `scripts/tasks/sandbox-parts.ts` to configure Vitest in sandboxes.
- Modified `scripts/task.ts` to include the new `vitest-integration` task.
- Added `scripts/tasks/vitest-test.ts` to run Vitest tests within sandboxes.
- Updated `code/lib/cli/src/sandbox-templates.ts` to integrate the new task into sandbox creation.
- Enhanced `code/renderers/react/src/preset.ts` to support experimental React Server Components (RSC).

### CODE REVIEW

File: scripts/tasks/sandbox-parts.ts [hunk 1]
```diff
@@ -363,6 +365,151 @@ async function linkPackageStories(
...
+          name: 'storybook',
+          include: [
+           // we need to set the path like this because svelte-kit overrides the root path so this makes it work in all sandboxes
+            path.join(__dirname, '../src/**/*.{story,stories}.?(c|m)[jt]s?(x)'),
+            path.join(__dirname, '../template-stories/**/*.{story,stories}.?(c|m)[jt]s?(x)'),
+          ],
+          exclude: [
+            ...defaultExclude,
+            // TODO: investigate TypeError: mocked(...).mockReturnValue is not a function
+            path.join(__dirname, '../**/*module-mocking*'),
```
Comment (yannbf) [inline scripts/tasks/sandbox-parts.ts:445]: We need to set up the conditions in the plugin itself

File: code/frameworks/nextjs/src/portable-stories.ts [hunk 2]
```diff
@@ -16,6 +16,7 @@ import type {
 
 // ! ATTENTION: This needs to be a relative import so it gets prebundled. This is to avoid ESM issues in Nextjs + Jest setups
 import { INTERNAL_DEFAULT_PROJECT_ANNOTATIONS as reactAnnotations } from '../../../renderers/react/src/portable-stories';
+import * as rscAnnotations from '../../../renderers/react/src/entry-preview-rsc';
```
Comment (yannbf) [inline code/frameworks/nextjs/src/portable-stories.ts:19]: only include decorators, not the parameters

File: scripts/tasks/sandbox-parts.ts [hunk 3]
```diff
@@ -363,6 +365,150 @@ async function linkPackageStories(
...
+            path.join(__dirname, '../**/frameworks/sveltekit_svelte-kit-skeleton-ts/forms.stories*'),
+          ],
+          /**
+           * TODO: Either fix or acknowledge limitation of:
+           * - Storybook channel usage:
+           * -- KeydownDuringPlay, Hooks, Events, ChangeArgs
+           * - @storybook/core/preview-api hooks:
+           * -- UseState
+           * - Targeted args:
+           * -- Targets
```
Comment (yannbf) [inline scripts/tasks/sandbox-parts.ts:475]: we can skip these for now, targeted args are abandoned

File: .circleci/config.yml [hunk 4]
```diff
@@ -662,40 +682,44 @@ workflows:
       - script-checks:
           requires:
             - build
-      - chromatic-internal-storybook:
-          requires:
-            - build
+      # - chromatic-internal-storybook:
```
Comment (yannbf) [inline .circleci/config.yml:685]: undo this

**Review by kasperpeulen — APPROVED** [review]
LGTM

**Pull request merged**